src/scry/implementations.cr

```diff
@@ -34,10 +34,34 @@ module Scry
 
     # NOTE: compiler is a bit heavy in some projects.
     def search(filename, position)
+      local_require = get_local_require(filename, position)
+      return local_require if local_require
       scope = get_scope(@workspace.root_uri)
       analyze(filename, position, scope)
     end
 
+    LOCAL_REQUIRE_REGEX = /^require\s+"(\.{1,2}.*?)"\s*$/
+    POSITION_0          = Protocol::Position.new(0, 0)
+    RANGE_0             = Protocol::Range.new(POSITION_0, POSITION_0)
+
+    private def get_local_require(filename, position)
+      line = @text_document.get_line(position.line)
+      if match = line.match(LOCAL_REQUIRE_REGEX)
+        requirement = match[1]
+        start_pos = line.index(requirement)
```

Comment:
Not needed, the information is available in `match`!
`match.byte_begin(1)`
(the return below can then be removed) 

Same for `end_pos`, it can be written: `match.byte_end(1)`